gym/lib/gym/code_signing_mapping.rb

```diff
@@ -107,23 +107,45 @@ def detect_project_profile_mapping
       self.project_paths.each do |project_path|
         UI.verbose("Parsing project file '#{project_path}' to find selected provisioning profiles")
 
+        # Storing bundle identifiers with duplicate profiles
+        # for informing user later on
+        bundle_identifiers_with_duplicates = []
```

Comment:
That was me being a dingus 😊 I moved them inside the begin. 

And I have not tested it with multiple targets but my example has 1 target with 6 configurations which will behave the same as 2 targets with 3 configurations (theoretically)... But I will test 😇